### PR TITLE
Update readme.md

### PR DESCRIPTION
**Pull request by StDevon**

### CODE REVIEW

**Comment by papeye** [comment]
could you remove also:
```
Create venv
python -m venv venv
Activate venv
.\venv\Scripts\Activate.ps1
install requirements
pip install -r requirements.txt
```
from readme?

File: readme.md [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+## Functionality
+
+The package provides the class SpeakerIdentifier, which initializes a CNN for speaker identification. This class requires a path to a directory containing training audio files. The audio file names should represent the speakers' names:
 ```
-.\venv\Scripts\Activate.ps1
+user1 = SpeakerIdentifier(model_name="user1",training_ds_dir=train_example_dir)
 ```
 
-### install requirements
+ Note that in the current version, only single files are used for training and it is assumed that in each of the audio files, only one speaker is present. The audio files are subsequently processed by optional Voice Activity Detection (optional) to remove silence and split into 1s segments which are used for the training. Optional data augmentation is possible by adding noise to the training audios (configurable in Config file).
```
Comment (papeye) [inline readme.md:24]: Could you make "Note that in current version..." into a note:
> [!NOTE]  
> Only single file per speaker is supported for training at the moment

Comment (papeye) [reply readme.md:24]: and info about one speaker speaking in audio into 'important':
> [!IMPORTANT]  
> It is assumed that only one person is speaking in an audio file

File: readme.md [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
+
+The training is done via train method of SpeakerIdentifier:
 ```
-pip install -r requirements.txt
+user1.train(train_data_dir=train_example_dir, with_vad=True)
 ```
-### Example run on 20 speakers using 
+
+The number of epochs can be adjusted in the Config file, but on the example audio 90% accuracy is obtained for only 3 epochs (which is default). Convolutional neural network is defined in nnmodel.py file using residual blocks and its size is chosen for identifying 20 speakers but users are encouraged to experiment with hyperparameters for the best performance. If user provides less then 20 speakers, base speakers are added to the training dataset to achive the best consistent performance. However if user provides more than 20 speakers in the training_ds_dir no base speakers will be added to the training dataset. The trained model can be used to predict the speaker from the audio file by calling
```
Comment (papeye) [inline readme.md:31]: from the point of view of a user this info is additional since they cannot (in assumption) alter the library code. I would put this into "For contributors" section

File: readme.md [hunk 3]
```diff
@@ -1,27 +1,38 @@
+# Speaker identification
+
 Implementation of CNN for speaker identification in TensorFlow. Trained on the example data of 20 speakers, the network correctly identifies about 90% of the speakers. 
 
-# Running instructions
+## Running instructions
```
Comment (papeye) [inline readme.md:5]: I'd rather we change it into sections 'setup' wher is pip install command and  'usage' where we show how one should use this library on their data which would include codeblock inside readme that does what example.py does but in a generic form e.g.:
```
speaker_identifier= SpeakerIdentifier(model_name=<model_name>, training_ds_dir=<train_data_dir>)
```
and other instructions...

File: readme.md [hunk 4]
```diff
@@ -1,27 +1,38 @@
...
 To install the package run
 ```
 pip install git+https://github.com/papeye/speaker-identification.git@setup-package
 ```
 
-### Create venv
+### Example run on 20 speakers using 
 ```
-python -m venv venv
+python .\example.py
```
Comment (papeye) [inline readme.md:14]: we dont need to show how to run it. We can just mention that one could look into example.py for specific use case

File: readme.md [hunk 5]
```diff
@@ -1,27 +1,38 @@
...
-### Example run on 20 speakers using 
+
+The number of epochs can be adjusted in the Config file, but on the example audio 90% accuracy is obtained for only 3 epochs (which is default). Convolutional neural network is defined in nnmodel.py file using residual blocks and its size is chosen for identifying 20 speakers but users are encouraged to experiment with hyperparameters for the best performance. If user provides less then 20 speakers, base speakers are added to the training dataset to achive the best consistent performance. However if user provides more than 20 speakers in the training_ds_dir no base speakers will be added to the training dataset. The trained model can be used to predict the speaker from the audio file by calling
+
 ```
-python .\example.py
+predictions = user1.predict(test_data_dir=test_example_dir, with_vad=True)
 ```
+
+which returns a class mapping each speaker directory to a dictionary of predicted speaker labels and their normalized probabilities, sorted in descending order of probability.
```
Comment (papeye) [inline readme.md:37]: name the class model that is returned (Result) and mention getters it has so the user would know what the can access

File: readme.md (hunk 1 above)
Comment (papeye) [reply readme.md:24]: other :https://github.com/orgs/community/discussions/16925

File: readme.md [hunk 6]
```diff
@@ -1,27 +1,38 @@
...
 
-### Create venv
+### Example run on 20 speakers using 
 ```
-python -m venv venv
+python .\example.py
 ```
 
-### Activate venv
+## Functionality
```
Comment (papeye) [inline readme.md:17]: to general imo, we should split it into specific sections

**Review by papeye — CHANGES_REQUESTED** [review]

File: readme.md (hunk 5 above)
Comment (StDevon) [reply readme.md:37]: the getter is just the best_prediction, right?

**Comment by papeye** [comment]
LGTM

**Review by papeye — APPROVED** [review]

File: readme.md (hunk 5 above)
Comment (papeye) [reply readme.md]: right

**Comment by Ryczek** [comment]
Add example output that user can expect and comment what is correct and what is not correct?

"If the user provides less than 20 speakers, base speakers are added to the training dataset to match the number of 20 training speakers to achieve the best consistent performance. However, if the user provides more than 20 speakers in the training_ds_dir no base speakers will be added to the training dataset."

**Comment by StDevon** [comment]
> Add example output that user can expect and comment what is correct and what is not correct?

do you mean something like that:
audio: prediction
edzik.wav: edzik.wav
papeye.wav: papeye.wav
ryczek.wav: ryczek.wav
Speaker0.wav: Speaker0.wav
Speaker1.wav: Speaker1.wav
Speaker10.wav: Speaker10.wav
Speaker11.wav: Speaker11.wav
Speaker12.wav: Speaker12.wav
Speaker13.wav: Speaker13.wav
Speaker14.wav: Speaker14.wav
Speaker2.wav: Speaker2.wav
Speaker3.wav: Speaker3.wav
Speaker4.wav: Speaker0.wav
Speaker5.wav: Speaker5.wav
Speaker6.wav: Speaker2.wav
Speaker7.wav: Speaker0.wav
Speaker8.wav: Speaker8.wav
Speaker9.wav: Speaker9.wav
wszewko.wav: wszewko.wav

        Execution 258.85028052330017

        Train: 148.94039607048035

        Prepare train: 83.70438885688782

        Prepare test: 18.780694484710693

        Predict: 6.440445423126221
        

I'm not sure if that is really important....
